btsieve/src/bitcoin/bitcoind_connector.rs

```diff
@@ -41,7 +41,7 @@ impl LatestBlock for BitcoindConnector {
     type BlockHash = bitcoin_support::BlockId;
 
     fn latest_block(
-        &self,
+        &mut self,
```

Comment:
At the moment it is not needed but it was introduced for testing purposes (see implementation of `LatestBlock` for `BitcoinConnectorMock`). Since we can see future, smarter implementations of `LatestBlock` needing to modify state inside the blockchain connector we decided that it was okay.